.github/workflows/build-hoprd.yaml

```diff
@@ -0,0 +1,86 @@
+---
+name: Build Hoprd
+
+on:
+  workflow_call:
+    secrets:
+      GOOGLE_HOPRASSOCIATION_PROJECT:
+        required: true
+      GOOGLE_HOPRASSOCIATION_CREDENTIALS_REGISTRY:
+        required: true
+      K8S_SSH_SERVICE_HOST:
+        required: true
+      K8S_SSH_SERVICE_USERNAME:
+        required: true
+      K8S_SSH_SERVICE_PASSWORD:
+        required: true
+concurrency:
+  group: ${{ github.head_ref }}-hoprd
+  cancel-in-progress: true
+jobs:
+
+  build:
+    name: docker
+    runs-on: ubuntu-2-core
+    steps:
+      - name: Checkout hoprnet repository
+        uses: actions/checkout@v3          
+
+      - id: auth
+        name: Set up Google Cloud Credentials
+        uses: google-github-actions/auth@v1
+        with:
+          token_format: "access_token"
+          credentials_json: ${{ secrets.GOOGLE_HOPRASSOCIATION_CREDENTIALS_REGISTRY }}
+
+      - name: Login Google Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: europe-west3-docker.pkg.dev
+          username: oauth2accesstoken
+          password: ${{ steps.auth.outputs.access_token }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver: kubernetes
+
+      - name: "Set PR number on version"
+        run: ./scripts/bump-version.sh ${{ github.event.pull_request.number }}
+
+      - name: Build and push docker image
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          file: packages/hoprd/Dockerfile
+          cache-from: type=registry,ref=europe-west3-docker.pkg.dev/${{ secrets.GOOGLE_HOPRASSOCIATION_PROJECT }}/docker-images/hoprd:pr-${{ github.event.pull_request.number }}-cache
+          cache-to: type=registry,ref=europe-west3-docker.pkg.dev/${{ secrets.GOOGLE_HOPRASSOCIATION_PROJECT }}/docker-images/hoprd:pr-${{ github.event.pull_request.number }}-cache,mode=max
+          tags: europe-west3-docker.pkg.dev/${{ secrets.GOOGLE_HOPRASSOCIATION_PROJECT }}/docker-images/hoprd:pr-${{ github.event.pull_request.number }}
+
+  restart_deployment:
+    name: Restart deployment
+    runs-on: ubuntu-2-core
+    needs: build
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'deploy_nodes') }}
+    steps:
+      - name: "Setup environment"
+        run: |
+          prBase="${{ github.event.pull_request.base.ref }}"
+          network=$(jq -r "to_entries[] | select(.value.git_ref | test(\"${prBase/*\/}\")).value.network" packages/hoprd/releases.json)
+          echo "NETWORK_DASH=${network/_/-}" >> $GITHUB_ENV
+      - name: Executing remote SSH command
+        uses: appleboy/ssh-action@v0.1.10
```

Comment:
I consider using a GH action a security risk since that could change and basically expose our secrets. Why not simply invoke `ssh` directly?

Reply:
I considered a better option than to reinvent the wheel. I would need to review the security risks about using secrets on external actions, as we are doing it in many workflows, and this would be a general concern in the community about using actions not certified by Github.

Nevertheless this might change soon, probably in future PR, as now we have the nodes running in Kubernetes and we can execute the kubectl command directly without accessing thourgh ssh. For doing that, I would have to rebuild the base image used in github runners to add the `kubectl` command, and grant permissions to the service account using the github runner to restart deployments.